### PR TITLE
Cleanup and fix mistakes in libraries

### PR DESCRIPTION
**Pull request by PokeCodec**
I went ahead and refactored the library and m4a code so it looks better and they all match.

### CODE REVIEW

File: libagbsyscall/libagbsyscall.s [hunk 1]
```diff
@@ -145,22 +145,22 @@ HuffUnComp:
...
+	thumb_func_start SoftResetExram
 SoftResetExram:
 	ldr r3, =REG_IME
 	movs r2, #0
 	strb r2, [r3, #0]
 	ldr r3, =SOFT_RESET_DIRECT_BUF
 	movs r2, #1
 	strb r2, [r3, #0]
-	subs r3, #SOFT_RESET_DIRECT_BUF - 0x3007f00
+	sub r3, #SOFT_RESET_DIRECT_BUF - 0x3007f00
```
Comment (PikalaxALT) [inline libagbsyscall/libagbsyscall.s:156]: Why remove the condition suffices? (+#L159)

Comment (PokeCodec) [reply libagbsyscall/libagbsyscall.s:156]: Because this is syntax divided; thumb doesn't use that suffix

Comment (PikalaxALT) [reply libagbsyscall/libagbsyscall.s:156]: Then why leave the others (`movs`)?

Comment (PokeCodec) [reply libagbsyscall/libagbsyscall.s:156]: I fixed that too!

**Comment by GriffinRichards** [comment]
Build failure again. Bad find + replace

**Comment by PokeCodec** [comment]
Fixed!

File: constants/m4a_constants.inc [hunk 2]
```diff
@@ -6,194 +6,178 @@
 	.equiv TONEDATA_TYPE_FIX, 0x08
 	.equiv TONEDATA_TYPE_SPL, 0x40 @ key split
 	.equiv TONEDATA_TYPE_RHY, 0x80 @ rhythm
+	.equiv TONEDATA_P_S_PAN,  0xc0
 
-	.macro struct_begin
-	.struct 0
-	.endm
+	.equiv o_SoundChannel_status, 0
```
Comment (GriffinRichards) [inline constants/m4a_constants.inc:11]: Revert, same as https://github.com/pret/pokeruby/pull/796#discussion_r476373667

File: include/gba/m4a_internal.h [hunk 3]
```diff
@@ -140,55 +140,13 @@ struct SoundChannel
     struct WaveData *wav;
     u32 cp;
     struct MusicPlayerTrack *track;
-    u32 pp;
-    u32 np;
-    u32 d4;
+    void* pp;
+    void* np;
+    void* d4;
```
Comment (GriffinRichards) [inline include/gba/m4a_internal.h:145]: Same as https://github.com/pret/pokeruby/pull/796#discussion_r476374006

File: libagbsyscall/libagbsyscall.s [hunk 4]
```diff
@@ -1,9 +1,12 @@
 	.include "../constants/gba_constants.inc"
 	.include "../asm/macros/function.inc"
+	
+	.syntax unified
 
 	.text
 
 	.set SOFT_RESET_DIRECT_BUF, 0x03007FFA
+	.set USER_STACK,			0x03007F00
```
Comment (GriffinRichards) [inline libagbsyscall/libagbsyscall.s:9]: Alignment

File: src/AgbRfu_LinkManager.c [hunk 5]
```diff
@@ -352,7 +352,7 @@ static bool8 rfu_LMAN_linkWatcher(u16 REQ_commandID)
             {
                 for (i = 0; i < RFU_CHILD_MAX; i++)
                 {
-                    if ((bm_linkLossSlot >> i) & 1)
+                    if ((bm_linkLossSlot) & (1 << i))
```
Comment (GriffinRichards) [inline src/AgbRfu_LinkManager.c:355]: Unnecessary parens

File: src/m4a.c [hunk 6]
```diff
@@ -433,13 +432,10 @@ void m4aSoundMode(u32 mode)
         soundInfo->maxChans = temp >> SOUND_MODE_MAXCHN_SHIFT;
 
         temp = MAX_DIRECTSOUND_CHANNELS;
```
Comment (GriffinRichards) [inline src/m4a.c:434]: Assuming this is no longer necessary after below changes

File: src/m4a.c [hunk 7]
```diff
@@ -510,27 +497,26 @@ void m4aSoundVSyncOff(void)
...
-        CpuFill32(0, soundInfo->pcmBuffer, sizeof(soundInfo->pcmBuffer));
-    }
+    CpuFill32(0, soundInfo->pcmBuffer, sizeof(soundInfo->pcmBuffer));
 }
 
 void m4aSoundVSyncOn(void)
 {
     struct SoundInfo *soundInfo = SOUND_INFO_PTR;
-    u32 ident = soundInfo->ident;
+    const u32 ident = soundInfo->ident;
```
Comment (GriffinRichards) [inline src/m4a.c:519]: Unnecessary

File: libagbsyscall/libagbsyscall.s (hunk 4 above)
Comment (PokeCodec) [reply libagbsyscall/libagbsyscall.s:9]: they are aligned though. one with tabs, the other with spaces?

Comment (GriffinRichards) [reply libagbsyscall/libagbsyscall.s:9]: There's a big 3 tab gap here for no reason, either align it with `0x03007FFA` with spaces or just use a single space

**Comment by PokeCodec** [comment]
Fixed!

**Pull request merged**